spec/yearmonth.html

```diff
@@ -3,5 +3,511 @@
+        1. Return the new Record {
+          [[Year]]: _year_,
+          [[Month]]: _month_,
+          }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-YearMonthFromString" aoid="YearMonthFromString">
+      <h1>YearMonthFromString ( _yearMonth_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_yearMonth_) is String.
+        1. Let _result_ be ParseYearMonthString(_yearMonth_).
+        1. If ! ValidateYearMonth(_result_.[[Year]], _result_.[[Month]]) is *false*, then
+          1. Throw a *TypeError* exception.
+        1. Return ? CreateYearMonth(_result_.[[Year]], _result_.[[Month]]).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id=sec-temporal-ToYearMonth aoid=ToYearMonth>
+      <h1>ToYearMonth ( _yearMonth_ )</h1>
+      <emu-alg>
+        1. If Type(_yearMonth_) is String, then
+          1. Return ? YearMonthFromString(_yearMonth_).
+        1. If Type(_yearMonth_) is Object, then
+          1. If _yearMonth_ has an [[InitializedTemporalYearMonth]] internal slot, then
+            1. Return _yearMonth_.
+          1. Let _result_ be the new Record {
+            [[Year]]: *undefined*,
+            [[Month]]: *undefined*,
+            }.
+          1. For each row of <emu-xref href="#table-temporal-yearmonthlike-properties"></emu-xref>, except the header row, in table order, do
+            1. Let _property_ be the Property value of the current row.
+            1. Let _value_ be ? Get(_yearMonth_, _property_).
+            1. If _value_ is *undefined*, then
+              1. Throw a *TypeError* exception.
+            1. Let _value_ be ? ToInteger(_value_).
+            1. Set _result_'s internal slot whose name is the Internal Slot value of the current row to _value_.
+          1. If ! ValidateYearMonth(_result_.[[Year]], _result_.[[Month]]) is *false*, then
+            1. Throw a *TypeError* exception.
+          1. Return ? CreateYearMonth(_result_.[[Year]], _result_.[[Month]]).
+        1. Throw a *TypeError* exception.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id=sec-temporal-YearMonthToString aoid=YearMonthToString>
+      <h1>YearMonthToString ( _yearMonth_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_yearMonth_) is Object.
+        1. Assert: _yearMonth_ has an [[InitializedTemporalYearMonth]] internal slot.
+        1. Let _year_ be PadYear(_yearMonth_.[[Year]]).
+        1. Let _month_ be _yearMonth_.[[Month]] formatted as a two-digit decimal number, padded to the left with a zero if necessary.
+        1. Return the string-concatenation of _year_, the code unit 0x002D (HYPHEN-MINUS), and _month_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id=sec-temporal-YearMonthCompare aoid=YearMonthCompare>
+      <h1>YearMonthCompare ( _one_, _two_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_one_) is Object.
+        1. Assert: _one_ has an [[InitializedTemporalYearMonth]] internal slot.
```

Comment:
it seems like we should have an `IsYearMonth` abstract op, rather than constantly needing to assert that it's both an object and has the right internal slots.

Reply:
See https://github.com/tc39/proposal-temporal/issues/207.